mono/mini/aot-compiler.c

```diff
@@ -7163,6 +7169,9 @@ mono_aot_parse_options (const char *aot_options, MonoAotOptions *opts)
 			opts->mode = MONO_AOT_MODE_FULL;
 		} else if (str_begins_with (arg, "hybrid")) {
 			opts->mode = MONO_AOT_MODE_HYBRID;			
+		} else if (str_begins_with (arg, "interp")) {
+			opts->mode = MONO_AOT_MODE_FULL;
```

Comment:
We should have `opts->mode = MONO_AOT_MODE_INTERP;`, remove `opts->interp = TRUE;`, add `mono_aot_mode_is_interp ()`, and check for interp mode wherever we are checking for fullAOT mode and wherever you are checking for `acfg->aot_opts.interp`.